cloud/dimensiondata/dimensiondata_vlan.py

```diff
@@ -0,0 +1,411 @@
+    driver = DimensionData(user_id, key, region=region)
+
+    # Get Network Domain Object
+    network_domain = get_network_domain_by_name(driver, network_domain_name,
+                                                location)
+
+    # Process action
+    if action == 'create':
+        if name is False:
+            module.fail_json(msg="'name' is a required argument when action" +
+                                 " is 'create'")
+        res = create_vlan(module, driver, location, network_domain, name,
+                          description, base_address, prefix_size)
+        vlan = vlan_obj_to_dict(res['vlan'])
+        if res['changed'] is False:
+            module.exit_json(changed=False, msg="VLAN with name '%s'" +
+                             " already exists." % name, vlan=vlan)
+        module.exit_json(changed=True, msg="Successfully created VLAN.",
+                         vlan=vlan)
+    elif action == 'read' or action == 'get':
+        res = get_vlan(module, driver, location, network_domain,
+                       vlan_id, name)
+        if type(res) is dict:
+            vlan = res
+        else:
+            vlan = vlan_obj_to_dict(res)
+        module.exit_json(changed=False, msg="Successfully read VLAN.",
+                         vlan=vlan)
+    elif action == 'delete':
+        res = delete_vlan(module, driver, location, network_domain,
+                          vlan_id, name)
+        module.exit_json(changed=res, msg="Successfully deleted VLAN or " +
+                         "VLAN does not exist")
+    elif action == 'update':
+        if vlan_id is False:
+            module.fail_json(msg="'vlan_id' is a required argument when " +
+                             "updating a VLAN")
+        res = update_vlan(module, driver, location, network_domain, vlan_id,
+                          name, description)
+        if res['vlan'] is False and res['changed'] is False:
+            module.exit_json(changed=False, msg="VLAN did not need updating.")
+        else:
+            vlan = vlan_obj_to_dict(res['vlan'])
+            module.exit_json(changed=False, msg="Successfully updated VLAN.",
+                             vlan=vlan)
+    elif action == 'expand':
+        if vlan_id is False:
+            module.fail_json(msg="'vlan_id' is a required argument when " +
+                             "expanding a VLAN")
+        res = expand_vlan(module, driver, location, network_domain, vlan_id,
+                          name, prefix_size)
+        if type(res) is bool:
+            module.exit_json(changed=False, msg="VLAN already requested size.")
+        else:
+            module.exit_json(changes=True, msg="VLAN network size modified.",
+                             vlan=vlan_obj_to_dict(res))
+    else:
+        fail_json(msg="Requested action was " +
+                  "'%s'. Action must be one of 'create', 'read', " +
+                  "'update', 'expand' or 'delete'" % state)
```

Comment:
Since action is a choices field in the arg spec, this case shouldn't happen. (the arg_spec validation will fail_json() with a similar message to this if an invalid choice is specified.)